### ISSUE TITLE
Quite a few build warnings building recursor on Fedora rawhide

### ISSUE THREAD

**Issue by rubenk** [PowerDNS/pdns#6660]
- Program: Recursor
 - Issue type: Bug report

### Short description

Building the recursor on Fedora 28 generates quite a few build warnings like the following:
```
  CXX      dnssecinfra.o
dnsrecords.cc: In member function 'ComboAddress AAAARecordContent::getCA(int) const':
dnsrecords.cc:119:30: warning: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'union ComboAddress'; use assignment or value-initialization instead [-Wclass-memaccess]
   memset(&ret, 0, sizeof(ret));
                              ^
In file included from dnsrecords.cc:27:
iputils.hh:85:7: note: 'union ComboAddress' declared here
 union ComboAddress {
       ^~~~~~~~~~~~
```

It's not a big issue, but the amount of warnings makes it hard to see the important ones.

### Environment

 - Operating system: Fedora rawhide
 - Software version: recursor 4.1.3, g++ 8.1
 - Software source: tarball from powerdns.com

**Comment by pieterlexis** [comment]
#6590 might need backporting

**Comment by omoerbeek** [comment]
Merged and backported to 4.1
